libtransmission/rpcimpl.cc

```diff
@@ -1696,6 +1705,100 @@ static char const* torrentAdd(tr_session* session, tr_variant* args_in, tr_varia
 ****
 ***/
 
+static char const* groupGet(tr_session* s, tr_variant* args_in, tr_variant* args_out, struct tr_rpc_idle_data* /*idle_data*/)
+{
+    tr_variant* namesList = nullptr;
+    std::set<std::string_view> names;
+
+    if (std::string_view one_name; tr_variantDictFindStrView(args_in, TR_KEY_name, &one_name))
+    {
+        names.insert(one_name);
+    }
+    else if (tr_variantDictFindList(args_in, TR_KEY_name, &namesList))
+    {
+        int names_count = tr_variantListSize(namesList);
+        for (int i = 0; i < names_count; i++)
+        {
+            tr_variant* v = tr_variantListChild(namesList, i);
+            std::string_view l;
+            if (tr_variantIsString(v) && tr_variantGetStrView(v, &l))
+            {
+                names.insert(l);
+            }
+        }
+    }
+
+    tr_variant* list = tr_variantDictAddList(args_out, TR_KEY_group, 1);
+    for (auto const& [name, group] : s->bandwidth_groups)
+    {
+        if (names.empty() || names.count(name) > 0)
+        {
+            tr_variant* dict = tr_variantListAddDict(list, 5);
+            bandwidth_limits limits;
+            group->getLimits(&limits);
+            tr_variantDictAddStrView(dict, TR_KEY_name, name);
+            tr_variantDictAddBool(dict, TR_KEY_uploadLimited, limits.up_limited);
+            tr_variantDictAddInt(dict, TR_KEY_uploadLimit, limits.up_limit_KBps);
+            tr_variantDictAddBool(dict, TR_KEY_downloadLimited, limits.down_limited);
+            tr_variantDictAddInt(dict, TR_KEY_downloadLimit, limits.down_limit_KBps);
+            tr_variantDictAddBool(dict, TR_KEY_honorsSessionLimits, group->areParentLimitsHonored(TR_UP));
+        }
+    }
+
+    return nullptr;
+}
+
+static char const* groupSet(
+    tr_session* session,
+    tr_variant* args_in,
+    tr_variant* /*args_out*/,
+    struct tr_rpc_idle_data* /*idle_data*/)
+{
+    std::string_view name;
+
+    if (!tr_variantDictFindStrView(args_in, TR_KEY_name, &name))
+    {
+        return "No group name given";
+    }
+
+    Bandwidth* group = session->bandwidthGroupFind(name);
+    if (group == nullptr)
+    {
+        return "No such group";
+    }
+
+    bandwidth_limits limits;
+    group->getLimits(&limits);
+
+    tr_variantDictFindBool(args_in, TR_KEY_speed_limit_down_enabled, &limits.down_limited);
+    int64_t intVal = 0;
+    if (limits.down_limited && tr_variantDictFindInt(args_in, TR_KEY_speed_limit_down, &intVal))
```

Comment:
The speed limit can't be changed unless limiting is enabled. is this intentional?

```suggestion
    if (tr_variantDictFindInt(args_in, TR_KEY_speed_limit_down, &intVal))
```